### DISCUSSION TITLE
How do I mount an overlay over /sysroot?

### DISCUSSION THREAD

**Discussion by oddlama** [dracutdevs/dracut#1787] [Q&A]
I'm trying to create a small dracut module [[1]](https://github.com/oddlama/dracut-rootoverlay) that allows to mount a writable overlay over the whole root partition. The idea is not to create a live-stick environment, but to have a read-only /sysroot with a writable overlay on a second disk, specified separately like `rootoverlay=UUID=...`. This is useful when spinning up many VMs that share a common system installation but have to be configured slightly differently.

My problem is, I want to allow dracut to setup `/sysroot` as usual, using whatever it would use normally. Unfortunately, systemd doesn't allow to create a second mount unit with the same `Where=` as the already existing `sysroot.mount` unit, as the filename must match the `Where` option. So I cannot mount an overlayfs on top of /sysroot before switch-root occurs. I also cannot change where `/sysroot` gets mounted, because a lot of dracut modules hard-code `/sysroot` (e.g. ZFS).

So the only remaining option that I see is to mount the overlay to e.g. `/sysroot-overlay`, and tell dracut right after mounting it, that from that point `NEWROOT=/sysroot-overlay`, so that `/sys, /proc, ...` are mounted in this directory, and so the switchroot will occur correctly. Unfortunately I cannot find any documentation on how to achieve this rather specific request. I'd be happy about any pointers.

**Comment by LaszloGombos** [comment]
@oddlama  Have you had a chance to try rd.live.overlay=UUID=99440c1f-8daa-41bf-b965-b7240a8996f4 ? 

See https://man7.org/linux/man-pages/man7/dracut.cmdline.7.html#:~:text=live.overlay%3DUUID%3D-,99440c1f,-%2D8daa%2D41bf%2Db965

Also relevant - https://github.com/dracutdevs/dracut/pull/1744
